src/tools/pub/pubspec.md

```diff
@@ -176,10 +184,26 @@ public.
 This should be a URL pointing to the website for your package.
 For [hosted packages](/tools/pub/dependencies#hosted-packages),
 this URL is linked from the package's page.
-While this is technically optional *please do* provide one. It
-helps users understand where your package is coming from. If nothing else, you
-can always use the URL where you host the source code, such as
-[GitHub](https://github.com).
+While this is technically optional *please do* provide either a `homepage`
+or a `repository`. It helps users understand where your package is coming from.
+
```

Comment:
Both is OK, too, right? "either...or" makes it sound like you can't do both. How about:

While providing a `homepage` is optional, please provide it or `repository` (or both).
These fields help users understand...

Reply:
good point :)